enclave-runner/src/usercalls/mod.rs

```diff
@@ -1316,14 +1316,20 @@ impl<'tcs> IOHandlerInput<'tcs> {
 
     #[inline(always)]
     async fn wait(&mut self, event_mask: u64, timeout: u64) -> IoResult<u64> {
-        let wait = match timeout {
-            WAIT_NO => false,
-            WAIT_INDEFINITE => true,
-            _ => return Err(IoErrorKind::InvalidInput.into()),
-        };
-
         let event_mask = Self::check_event_set(event_mask)?;
 
+        let timeout = match timeout {
+            WAIT_NO | WAIT_INDEFINITE => timeout,
+            _ => {
+                // tokio::time::timeout does not handle large timeout values
+                // well which may or may not be a bug in tokio, but it seems to
+                // work ok with timeout values smaller than 2 ^ 55 nanoseconds.
+                // TODO: figure out if a bug report is in order.
```

Comment:
Ideally each TODO is accomanied by a link to some issue. Please file a bug report if there isn't one already.